include/basic_types.h

```diff
@@ -9,6 +9,20 @@
 #include <stdint.h>
 #include <arch/types.h>
 
+/* arch/types.h is supposed to define word_t and _seL4_word_fmt */
+#ifndef _seL4_word_fmt
```

Comment:
Have you tried just including <sel4/basic_types.h> from kernel source files?  libsel4 include directories are in the kernel include path.  Some of them aren't supported for inclusion in the kernel, but basic_types.h may work.

Reply:
ok, will give this a try since nobody has raised an objection in the last 4 days.

Reply:
This is probably not the best place to discuss, however, is include seL4 library header files in the kernel a good practise?

When first looking at the kernel I found it surprising, and made things more difficult to understand and debug. I guess is it something we really want to continue encouraging or eventually try to phase out?

Reply:
I don't really have an opinion here, but so far this seems pretty much separated. Obviously, merging this PR without change would be the easiest thing - and then discuss further steps. I've seen you also did some refactoring in your branches on the repo, which may eventually have some impact on this discussion?

Reply:
Yeah, the comment wasn't intended to say this shouldn't be merged, just to spark thinking.